### PR TITLE
Add script to run profilers on benchmarks

### PR DESCRIPTION
**Pull request by QRemy**
Add script to run profilers on benchmarks, it would be nice to have an action to run this at each merge in gammapy.

### CODE REVIEW

**Comment by QRemy** [comment]
This requires https://github.com/gammapy/gammapy/pull/4398

**Comment by adonath** [comment]
Have you had the chance to evaluate https://asv.readthedocs.io/en/stable/ for this task? This is what I used in Astropy, Scipy and Numpy.

File: benchmarks/profilers/explore_git_history.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+if __name__ == "__main__":
+   logging.basicConfig(level=logging.INFO)
+   
+   benchmarks = "../../"
+   files = glob.glob("../*.py")
+   benchs = [Path(f).stem for f in files if Path(f).stem != "make"]
+
+   for bench in benchs:
+       print(bench)
```
Comment (registerrier) [inline benchmarks/profilers/explore_git_history.py:146]: print or log?

File: benchmarks/profilers/explore_git_history.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+       try:
+           dates, bench_times, labels, commits = get_benchmark_history(bench, benchmarks)
+           mask = plot_benchmark_history(bench, dates, bench_times, labels)
+           ind_peak = np.where(mask)[0]
+           for ind in ind_peak:
+               print("Peak : ", dates[ind])
```
Comment (registerrier) [inline benchmarks/profilers/explore_git_history.py:152]: same here log (info?)

**Review by registerrier — APPROVED** [review]
Thanks @QRemy . See inline comments.

**Review by registerrier — APPROVED** [review]
Thanks @QRemy ! No more comments from my side.

**Pull request merged**